app/forms/waste_carriers_engine/payment_summary_form.rb

```diff
@@ -3,10 +3,14 @@
 module WasteCarriersEngine
   class PaymentSummaryForm < ::WasteCarriersEngine::BaseForm
     delegate :temp_payment_method, to: :transient_registration
+    delegate :receipt_email, to: :transient_registration, allow_nil: true
 
     attr_accessor :type_change, :registration_cards, :registration_card_charge, :total_charge
 
     validates :temp_payment_method, inclusion: { in: %w[card bank_transfer] }
+    validates :receipt_email, "defra_ruby/validators/email": true, if: :paying_by_card?
```

Comment:
I would argue that since we are pre-populating, we should always validate this. If a user selects bank transfer, they will still be submitting a value and we will still be saving it.

If they haven't selected WorldPay first and seen the form, this will always be their already validated contact email, and so there shouldn't be any issues.

However, if they select WorldPay, enter an invalid email and then switch back to bank transfer and submit, we will still be storing that value, only without any validation.

Alternatively I think we should wipe any receipt emails that are submitted in combination with the bank transfer option so we don't store any dodgy data.

Reply:
I had considered this scenario. I was wary of forcing a validation because I didn't want to get into the edge case of trying to force a hidden section to reveal if there is a validation error in a field that is no longer pertinent to them.

I was edging on
- likelihood of it happening
- could always be fixed when in the Edit screen (once I have implemented it)

And hoping no one would notice 😁 

Wiping it for bank transfer then becomes the alternative. But that also has an edge case scenario.

- User registers and pays by card so receipt email stored
- User renews and pays by bank transfer. We wipe the receipt email

Appreciate there is 3 years between these things happening so we would have time to actually implement receipts properly and this design would go away! But never say never!

So my current plan is to go back to my first commit idea and create an attribute on the form, pre-populated by whatever is in the transient_registration.

We then still only validate it if the payment type is card. But we also only pass it to be saved to the transient_registration if the payment type is card.

That way we can properly ignore it if payment type is bank transfer. But we also only overwrite any existing value if they are paying by card.

What do you think?

Reply:
That sounds like a good solution to me! :)